src/server/pps/server/pipeline_controller.go

```diff
@@ -191,19 +191,64 @@ func (pc *pipelineController) step(timestamp time.Time) (retErr error) {
 		tracing.FinishAnySpan(span, "err", retErr)
 	}()
 
+	if step, ctx, err := pc.newStep(); err != nil {
+		// if we fail to create a new step, there was an error querying the pipeline info, and there's nothing we can do
+		log.Errorf("PPS master: failed to set up step data to handle event for pipeline '%s': %v", pc.pipeline, errors.Wrapf(err, "failing pipeline %q", pc.pipeline))
+		return false, err
+	} else if step == nil {
+		// interpret the event as a delete operation
+		if err := pc.deletePipelineResources(); err != nil {
+			log.Errorf("PPS master: error deleting pipelineController resources for pipeline '%s': %v", pc.pipeline,
+				errors.Wrapf(err, "failing pipeline %q", pc.pipeline))
+			return true, err
+		}
+		return true, nil
+	} else {
+		var stepErr stepError
```

Comment:
this is correct, just needed for unwrapping in the `errors.As`. I guess it could be inline in the ifs 
`if stepErr := new(stepError); errors.As(err, &stepErr)`
or just have two declared closer to where it's used, but the interface for unwrapping is just a little annoying

Reply:
Got it, now I see how this works—thanks!!